datasetinsights/estimators/siam_mask.py

```diff
@@ -0,0 +1,2251 @@
+    h = y2 - y1
+    return x, y, w, h
+
+
+def center2corner(center):
+    """
+    Converts a corner representation to center representation
+    cx,cy,w,h -> x1,y1,x2,y2
+    :param center: Center or np.array 4*N
+    :return: Corner or np.array 4*N
+    """
+    x, y, w, h = center[0], center[1], center[2], center[3]
+    x1 = x - w * 0.5
+    y1 = y - h * 0.5
+    x2 = x + w * 0.5
+    y2 = y + h * 0.5
+    return x1, y1, x2, y2
+
+
+def anyform2canonicalBBox(form, rep_type):
+    """
+    Helper function for form conversion
+    Converting any form corner to center or opposite
+    """
+    if rep_type == "corner":
+        x1, y1, x2, y2 = form[0], form[1], form[2], form[3]
+        w, h = x2 - x1, y2 - y1
+
+    elif rep_type == "center":
+        corner = center2corner(form)
+        x1, y1, x2, y2 = corner[0], corner[1], corner[2], corner[3]
+        w, h = x2 - x1, y2 - y1
+
+    return BBox2D(label=None, x=x1, y=y1, w=w, h=h)
+
+
+def toBBox(image, shape, template_size=127):
+    """
+    Converting coordinates to standard BBox implementation
+    """
+    imh, imw = image.shape[:2]
+    if len(shape) == 4:
+        w, h = shape[2] - shape[0], shape[3] - shape[1]
+    else:
+        w, h = shape
+    context_amount = 0.5
+    exemplar_size = template_size
+    wc_z = w + context_amount * (w + h)
+    hc_z = h + context_amount * (w + h)
+    s_z = np.sqrt(wc_z * hc_z)
+    scale_z = exemplar_size / s_z
+    w = w * scale_z
+    h = h * scale_z
+    cx, cy = imw // 2, imh // 2
+    bbox = anyform2canonicalBBox([cx, cy, w, h], "center")
+    return bbox
+
+
+def crop_hwc(image, bbox, out_sz, padding=(0, 0, 0)):
+    """
+    Cropping image given a bounding box
+    Args: Image and modified bounding box
+    Returns: Cropped Image
+    """
+    x1, y1, w, h = bbox.x, bbox.y, bbox.w, bbox.h
+    bbox = [x1, y1, x1 + w, y1 + h]
+    bbox = [float(x) for x in bbox]
+    a = (out_sz - 1) / (bbox[2] - bbox[0])
+    b = (out_sz - 1) / (bbox[3] - bbox[1])
+    c = -a * bbox[0]
+    d = -b * bbox[1]
+    mapping = np.array([[a, 0, c], [0, b, d]]).astype(np.float)
+    crop = cv2.warpAffine(
+        image,
+        mapping,
+        (out_sz, out_sz),
+        borderMode=cv2.BORDER_CONSTANT,
+        borderValue=padding,
+    )
+    return crop
+
+
+def draw(image, box, name):
+    """
+        Args: image, bbox
+        Saves an image with bbox drawn on it
+        Only for debugging-> Not used anywhere
+    """
+    if box is not None:
+        box = [box.x, box.y, box.x + box.w, box.y + box.h]
+        box = [round(x) for x in box]
+    else:
+        box = [0, 0, 0, 0]
+    x1, y1, x2, y2 = int(box[0]), int(box[1]), int(box[2]), int(box[3])
+    image = image.copy()
+    # https://github.com/opencv/opencv/issues/14866#issuecomment-504632481 weird
+    cv2.rectangle(image, (x1, y1), (x2, y2), (0, 255, 0))
+    cv2.imwrite(name, image)
+
+
+"""
+    Generates a random number between 2 given numbers
+"""
+
+
+def genarate_random_number(a, b):
```

Comment:
Is this an integer sampled uniformly? if so, let's name this more clearly/explicitly.